feature_engine/creation/binary_combination_feature_reference.py

```diff
@@ -0,0 +1,250 @@
+from typing import List, Optional, Union
+
+import pandas as pd
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.utils.validation import check_is_fitted
+
+from feature_engine.dataframe_checks import (
+    _check_contains_na,
+    _check_input_matches_training_df,
+    _is_dataframe,
+)
+from feature_engine.variable_manipulation import _find_or_check_numerical_variables
+
+
+class BinaryCombinatioByFeatureReference(BaseEstimator, TransformerMixin):
+    """
+    BinaryCombinatioByFeatureReference() applies binary operations across targert and reference features,
+    returning 1 or more additional features as a result.
+
+    For example ...
+
+    Parameters
+    ----------
+
+    Reference_variables: list
+        The list of numerical reference variables to operate.
+
+    variables_to_combine: list
+        The list of target numerical variables to be combined.
+
+    binary_operations: list, default=None
+        The list of basic binary operations to be used in transformation.
+
+        If none, all of ['sub', 'div'] will be performed
+        over the variables. Alternatively, user can enter the list of operations to
+        carry out.
+
+        Each operation should be a string and must be one of the elements
+        from the list: ['sub', 'div']
+
+        Each operation will result in a new variable that will be added to the
+        transformed dataset.
+
+    new_variables_names: list, default=None
+        Names of the newly created variables. The user can enter a name or a list
+        of names for the newly created features (recommended). User must enter
+        one name for each binary transformation indicated in the math_operations
+        attribute. That is, if you want to perform mean and sum of features, you
+        should enter 2 new variable names. If you perform only mean of features,
+        enter 1 variable name. Alternatively, if you chose to perform all
+        binary transformations, enter 6 new variable names.
+
+        The name of the variables indicated by the user should coincide with the order
+        in which the binary operations are initialised in the transformer.
+        That is, if you set math_operations = ['mean', 'prod'], the first new variable
+        name will be assigned to the mean of the variables and the second variable name
+        to the product of the variables.
+
+        If new_variable_names=None, the transformer will assign an arbitrary name
+        to the newly created features starting by the name of the binary
+        operation, followed by the variables combined separated by -.
+
+    """
+
+    def __init__(
+        self,
+        variables_to_combine: List[Union[str, int]],
+        reference_variables: List[Union[str, int]],
+        binary_operations: [List[str]] = "sub",
```

Comment:
sub is not a list, and perhaps the default should be None?

See [here](https://github.com/solegalli/feature_engine/blob/master/feature_engine/creation/mathematical_combination.py#L224)

Reply:
Fixed